core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/SearchResultGenerator.kt

```diff
@@ -40,15 +40,24 @@ class ZimSearchResultGenerator @Inject constructor(
     if (sharedPreferenceUtil.prefFullTextSearch)
       zimReaderContainer.search(it, 200).run { fullTextResults() }
     else
-      zimReaderContainer.searchSuggestions(it, 200).run { suggestionResults() }
+      zimReaderContainer.searchSuggestions(it, 200).run { suggestionResults(it) }
 
   private fun fullTextResults() = generateSequence {
     zimReaderContainer.getNextResult()?.title?.let(::ZimSearchResultListItem)
   }.filter { it.value.isNotBlank() }
     .toList()
 
-  private fun suggestionResults() = generateSequence {
-    zimReaderContainer.getNextSuggestion()?.let { ZimSearchResultListItem(it.title) }
-  }.distinct()
-    .toList()
+  private fun suggestionResults(searchTerm: String) =
+    generateSequence {
+      zimReaderContainer.getNextSuggestion()?.let { ZimSearchResultListItem(it.title) }
+    }.distinct()
+      .toList()
+      .run {
```

Comment:
let might be more applicable?

Reply:
I get `Implicit parameter 'it' of enclosing lambda is shadowed` on the line with `none` if I do that

Reply:
and an `alt+enter` autocorrect allows you to explicitly name the parameter

Reply:
Moved to `State.Results`